.github/workflows/test.yml

```diff
@@ -1,45 +1,58 @@
-name: Spring Boot tests
+name: e2e tests
 
 on: push
 
 jobs:
-  microservices:
+  spring-boot-microservices:
     runs-on: ubuntu-latest
 
     steps:
       - name: Download repository
         uses: actions/checkout@v2
 
-      - name: Set up JDK 17
-        uses: actions/setup-java@v2
-        with:
-          java-version: '17'
-          distribution: 'adopt'
-
-      - name: Test place-service
-        run: cd "Spring Boot/microservices/place-service/" && make test
-        
-      - name: Test distance-service
-        run: cd "Spring Boot/microservices/distance-service/" && make test
-        
-      - name: Test route1-service
-        run: cd "Spring Boot/microservices/route1-service/" && make test
-        
-      - name: Test route2-service
-        run: cd "Spring Boot/microservices/route2-service/" && make test
-
-  monolith:
+      - name: Build docker environment
+        run: docker-compose -f docker-compose-spring-boot-microservices.yml up --build -d
+
+      - run: pip install -r .github/e2e/requirements.txt
```

Comment:
Rethink of taking out those 3 steps into some shared variable - there is some code redundancy here

Reply:
No solution found